deepchecks/utils/abstracts/confusion_matrix_abstract.py

```diff
@@ -27,6 +30,9 @@ def run_confusion_matrix_check(y_pred: np.ndarray, y_true: np.ndarray, with_disp
     total_classes = sorted([str(x) for x in set(y_pred).union(set(y_true))])
     result = confusion_matrix(y_true, y_pred)
 
+    # For accessing the class names from the condition
+    result = pd.DataFrame(result, index=total_classes, columns=total_classes)
```

Comment:
Do this after the if condition otherwise the create_confusion_matrix_figure function will fail